specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/workbooks_API.json

```diff
@@ -0,0 +1,583 @@
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The workbook link has been successfully deleted."
+          }
+        },
+        "x-ms-examples": {
+          "WorkbookList": {
+            "$ref": "./examples/WorkbookResponseOnly.json"
+          }
+        }
+      }
+    }    
+  },
+  "definitions": {
+    "Resource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "description": "Azure resource name",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "tags": {
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "x-ms-azure-resource": true,
+      "description": "An azure resource object"
+    },    
+    "Workbook": {
+      "description": "An Application Insights workbook definition.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "type": "string",          
+          "description": "The kind of workbook. Choices are user and shared.",
+          "enum": [
+            "user",
+            "shared"
+          ],
+          "x-ms-enum": {
+            "name": "SharedTypeKind",
+            "modelAsString": false
+          }
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Metadata describing a web test for an Azure resource.",
+          "$ref": "#/definitions/WorkbookProperties"
+        }
+      }
+    },
+    "WorkbookProperties": {
+      "description": "Properties that contain a workbook.",
+      "required": [
+        "name",
+        "kind",
+        "category",
+        "workbookId",
+        "serializedData",
+        "userId"
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The user-defined name of the workbook."
+        },
+        "serializedData": {
+          "type": "string",
+          "description": "Configuration of this particular workbook. Configuration data is a string containing valid JSON"
+        },
+        "version": {
+          "type": "string",
+          "description": "This instance's version of the data model. This can change as new features are added that can be marked workbook."
+        },
+        "workbookId": {

```

Comment:
How is the customer going to use this internal ID? Is it needed to surface it back to the user?